### ISSUE TITLE
`execute-dir` does not work when `_quarto.yml.local` exists

### ISSUE THREAD

**Issue by psychelzh** [quarto-dev/quarto-cli#8256]
### Bug description

For website project, if I set `execute-dir` as `"project"`, and I have also a copy of `_quarto.yml.local`, the working directory will be reset to `"file"`.

**Comment by mcanouil** [comment]
Could you share a working reproducible example?
Where did you set `execute-dir`? What's in your files? Etc.
We need to have all the information required to reproduce your issue.
For example, you could share a Git repository with the faulty setup.

Also, we asked different information in the issue template. It is not to annoy user but because we actually need those information.

**Comment by psychelzh** [comment]
I am sorry, maybe the local configuration file is not the cause of this issue. For my case, I have [a website project](https://github.com/psychelzh/cogstruct-dev) that when using interactively in RStudio, the `execute-dir` rule would not work recently, which I can confirm it worked formerly. However, if I use `quarto render`, it also works.

**Comment by psychelzh** [comment]
Interestingly, if I remove the label line `#| label: setup`, it also works interactively in RStudio.

**Comment by psychelzh** [comment]
I updated my post, and found the issue is related to the setup label.

**Comment by mcanouil** [comment]
I cannot reproduce on latest development version
<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/0a7a120c-a3b6-422f-94e2-52e8d138e3a3">

**Comment by psychelzh** [comment]
@mcanouil Yes, you are right. But I mean this only happens when I use it interactively in RStudio.

**Comment by mcanouil** [comment]
What do you mean exactly by "interactive"?

- When you execute the cell and not the document inside RStudio?
- When you Preview inside RStudio?

This starts to feel like an RStudio problem rather than a Quarto problem.

**Comment by psychelzh** [comment]
@cscheid Yes, I mean the first case, i.e., execute the cell and not the document inside RStudio.

**Comment by mcanouil** [comment]
So this is an RStudio issue.

**Comment by psychelzh** [comment]
Thanks! I am sorry for tagging at. Should I transfer this issue to RStudio? Actually, I think it worked in an older version of RStudio.

**Comment by mcanouil** [comment]
Note that in VSCode, what defines the working directory is the root of the workspace and not the document workspace, so for short in VSCode when executing "run cell" means send the code to the current terminal in whatever working directory it is (usually the root of the workspace/project).

**Comment by cderv** [comment]
I can reproduce on RStudio IDE 2024.04.0-daily+230 on Windows. 

Indeed #11370 seems to be the issue that made it possible, but it seems there is a missing handling of setup chunk which are special in RStudio IDE. 

@psychelzh can you open your issue there? 

Thanks !

**Comment by psychelzh** [comment]
Okay, I am going to cross post this in rstudio.
